modules/sdk-coin-btc/src/lib/keyPair.ts

```diff
@@ -0,0 +1,96 @@
+import { randomBytes } from 'crypto';
+import * as bip32 from 'bip32';
+import {
+  AddressFormat as BaseAddressFormat,
+  DefaultKeys,
+  isPrivateKey,
+  isPublicKey,
+  isSeed,
+  KeyPairOptions,
+  NotImplementedError,
+  Secp256k1ExtendedKeyPair,
+} from '@bitgo/sdk-core';
+import { address, crypto, networks } from 'bitcoinjs-lib';
+
+const DEFAULT_SEED_SIZE_BYTES = 16;
+
+// TODO: extend the AddressFormat enum in @bitgo/sdk-core with these values
+enum ExtendedAddressFormat {
+  mainnetBech32 = 'mainnetBech32',
+  testnetBech32 = 'testnetBech32',
+}
+
+export type AddressFormatType = BaseAddressFormat | ExtendedAddressFormat;
+export const AddressFormat = { ...BaseAddressFormat, ...ExtendedAddressFormat };
+
+/** Bitcoin key management. */
+export class KeyPair extends Secp256k1ExtendedKeyPair {
+  /**
+   * Public constructor. By default, creates a key pair with a random master seed.
+   *
+   * @param { KeyPairOptions } source Either a master seed, a private key (extended or raw), or a public key
+   *     (extended, compressed, or uncompressed)
+   */
+  constructor(source?: KeyPairOptions) {
+    super(source);
+    if (!source) {
+      const seed = randomBytes(DEFAULT_SEED_SIZE_BYTES);
+      this.hdNode = bip32.fromSeed(seed);
+    } else if (isSeed(source)) {
+      this.hdNode = bip32.fromSeed(source.seed);
+    } else if (isPrivateKey(source)) {
+      this.recordKeysFromPrivateKey(source.prv);
+    } else if (isPublicKey(source)) {
+      this.recordKeysFromPublicKey(source.pub);
+      // TODO: move this to Secp256k1ExtendedKeyPair
+      this.keyPair.compressed = source.pub.length === 66;
+    } else {
+      throw new Error('Invalid key pair options');
+    }
+
+    if (this.hdNode) {
+      this.keyPair = Secp256k1ExtendedKeyPair.toKeyPair(this.hdNode);
+    }
+  }
+
+  /**
+   * Bitcoin uncompressed public and private keys in hex format.
+   *
+   * @returns { DefaultKeys } The keys in the protocol default key format
+   */
+  getKeys(): DefaultKeys {
+    return {
+      pub: this.getPublicKey({ compressed: false }).toString('hex'),
```

Comment:
why do we want to support uncompressed keys?

Reply:
Looks like other coins that implement this method default to `compressed: false` but I also see some that accept `compressed` as a parameter to `getKeys`. We could do that here and have it default to `true`? That way we can "support" uncompressed keys but default compressed, which seems like the more sensible default.

Reply:
Changing default to compressed = true

Reply:
Changing the default to be `this.keyPair.compressed`